abdm/tasks/retry_failed_care_contexts.py

```diff
@@ -0,0 +1,38 @@
+from celery import shared_task
+from django.db.models import Q
+
+from abdm.models.abha_number import AbhaNumber
+from abdm.models.transaction import Transaction, TransactionStatus, TransactionType
+from abdm.service.v3.gateway import GatewayService
+
+
+@shared_task
+def retry_failed_care_contexts():
+    unsuccessfull_care_context_transactions = Transaction.objects.filter(
+        ~Q(status=TransactionStatus.COMPLETED),
```

Comment:
this would include, cancelled, failed and initiated as well

seems like in link__carecontext we are not updating it to failed if there is an error,
we could instead set status to failed if there is some error other than rate limit or connection issues

and only filter the txn here if the status is still as initiated

and set the status to failed here as well if there is a similar issue as above

Reply:
keeping this minimal for care v2, the same logic is copied over the care v3, will expand it more thoroughly there